staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -447,6 +376,205 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 	}
 }
 
+func (r *Reflector) list(stopCh <-chan struct{}) (string, error) {
```

Comment:
There is a bunch of refactoring in this PR (e.g. splitting this into a separate function itself (but there are multiple other things).

Can you please split them into a separate PR that we can merge earlier and make the actual changes smaller (and thus easier to follow)?

Reply:
[I would like to wait with the actual review before that's done as it would save a lot of my time.]